addons/fauxC7/CfgWeapons.hpp

```diff
@@ -0,0 +1,76 @@
+class CfgWeapons {
+    class ItemCore;
+    class InventoryMuzzleItem_Base_F;
+    class Rifle;
+    class Rifle_Base_F: Rifle {
+        class WeaponSlotsInfo;
+        class GunParticles;
+    };
+    class CUP_arifle_M16_Base: Rifle_Base_F {
+    };
+    class CUP_arifle_M16A4_Base: CUP_arifle_M16_Base {
+    };
+    class CUP_arifle_M16A4_GL: CUP_arifle_M16_Base {
+    };
+    class CUP_arifle_XM16E1: CUP_arifle_M16_Base {
+    };
+    class CUP_arifle_M16A1E1: CUP_arifle_XM16E1 {
+    };
+    class CUP_arifle_M16A1GL: CUP_arifle_XM16E1 {
+    }; 
+    class GVAR(c7): CUP_arifle_M16A1E1 {
+        author = "Potato";
+        displayName = "C7";
+        modes[] = {"Single","FullAuto","Burst_medium","single_medium_optics1","single_medium_optics2"};
+        class FullAuto: Mode_FullAuto {
```

Comment:
why are we redefining this?

Reply:
IIRC, it was because the M16A1E1 was burst fire and so in order to change that to be more accurate, it had to be redefined... I think. Don't quote me on that lol.

Reply:
Did you add this, or was it copy/pasted? As it is currently configured, the AI will go full-auto at a range of 300 meters

Reply:
Copy/pasted from the CUP M16A1 config found in the same file, iirc.

Reply:
Okay, did you test this in-game? I am not 100% sure this works, but I will need to investigate (if it doesnt work in game)

Reply:
I have not tested this with AI - I didn't even think about it being used by an AI haha

It should work for players at least, all it did was inherit the existing model, renamed it, and converted it to full auto. It'd probably be a good idea to double check it tho

Reply:
I am more talking about the inheritance of `Mode_fullAuto`; i don't know if this will be defined since the base gun is a 3 round burster

Reply:
I got this working, just needed to add sound configs